test/boost/frozen_mutation_test.cc

```diff
@@ -56,18 +55,28 @@ static future<> _test_freeze_unfreeze(UnfreezeFunc&& unfreeze_func) {
 }
 
 SEASTAR_TEST_CASE(test_writing_and_reading) {
-    return _test_freeze_unfreeze([] (const frozen_mutation& fm, schema_ptr s) {
+    return test_freeze_unfreeze(
+      [] (const mutation& m) {
+        return freeze(m);
+      },
+      [] (const frozen_mutation& fm, schema_ptr s) {
         return fm.unfreeze(std::move(s));
-    });
+      }
+    );
 }
 
 SEASTAR_TEST_CASE(test_writing_and_reading_gently) {
-    return _test_freeze_unfreeze([] (const frozen_mutation& fm, schema_ptr s) {
+    return test_freeze_unfreeze(
+      [] (const mutation& m) {
+        return freeze_gently(m).get();
+      },
+      [] (const frozen_mutation& fm, schema_ptr s) {
         return fm.unfreeze_gently(std::move(s)).handle_exception([] (std::exception_ptr ex) {
             testlog.error("test_writing_and_reading_gently: {}", ex);
             return make_exception_future<mutation>(std::move(ex));
         }).get();
-    });
+      }
+    );
```

Comment:
The new tests miss interoperability between gentle an ungentle versions. We could have freeze/unfreeze work, and freeze_gently/unfreeze_gently work, but have different serialized formats that don't interoperate.

Reply:
will fix in next version

Reply:
fixed in v5